dvc/commands/data.py

```diff
@@ -0,0 +1,106 @@
+import argparse
+import logging
+import os
+from contextlib import contextmanager
+from functools import partial
+
+from funcy import log_durations
+
+from dvc.cli.command import CmdBase
+from dvc.cli.utils import fix_subparsers
+from dvc.lock import LockNoop
+from dvc.ui import ui
+
+logger = logging.getLogger(__name__)
+
+print_durations = partial(
+    log_durations,
+    ui.error_write
+    if logger.isEnabledFor(logging.TRACE)  # type: ignore[attr-defined]
+    else logger.trace,  # type: ignore[attr-defined]
+)
+
+
+class CmdDataStatus(CmdBase):
+    @contextmanager
+    def _patch_clone(self):
+        from funcy import monkey
+
+        from dvc import scm
+
+        @monkey(scm, "clone")
+        def clone(url, *args, **kwargs):
+            with print_durations(f"cloning {os.path.basename(url)}"):
+                return clone.original(url, *args, **kwargs)
+
+        try:
+            yield
+        finally:
+            scm.clone = clone.original
+
+    @contextmanager
+    def _patched_lock_repo(self):
+        from dvc.repo import lock_repo
+
+        # rwlock is not atomic, so we may still want to have a repo lock here
```

Comment:
Our `rwlock` is not atomic enough, it can be easily demonstrated in `example-get-started` repo with following command:
```command
for i in $(seq 100); do
  dvc data status &
done; wait
```